tendrl/commons/flows/expand_cluster_with_detected_peers/__init__.py

```diff
@@ -0,0 +1,123 @@
+import time
+import uuid
+
+from tendrl.commons import flows
+from tendrl.commons.flows.exceptions import FlowExecutionFailedError
+from tendrl.commons.objects.job import Job
+from tendrl.commons.utils import log_utils as logger
+
+
+class ExpandClusterWithDetectedPeers(flows.BaseFlow):
+    def __init__(self, *args, **kwargs):
+        super(
+            ExpandClusterWithDetectedPeers,
+            self
+        ).__init__(*args, **kwargs)
+
+    def run(self):
+        integration_id = self.parameters['TendrlContext.integration_id']
+        _cluster = NS.tendrl.objects.Cluster(
+            integration_id=integration_id
+        ).load()
+        if _cluster.status is not None and _cluster.status != "" and \
+            _cluster.status in ["importing", "unmanaging", "expanding"]:
+            raise FlowExecutionFailedError(
+                "Another job in progress for cluster, please wait till "
+                "the job finishes (job_id: %s) (integration_id: %s) " % (
+                    _cluster.current_job['job_id'],
+                    integration_id
+                )
+            )
+        _lock_details = {
+            'node_id': NS.node_context.node_id,
+            'fqdn': NS.node_context.fqdn,
+            'tags': NS.node_context.tags,
+            'type': NS.type,
+            'job_name': self.__class__.__name__,
+            'job_id': self.job_id
+        }
+        _cluster.locked_by = _lock_details
+        _cluster.status = "expanding"
+        _cluster.current_job = {
+            'job_id': self.job_id,
+            'job_name': self.__class__.__name__,
+            'status': 'in_progress'
+        }
+        _cluster.save()
+
+        node_ids = self.parameters.get("Node[]", [])
```

Comment:
I thought detection code would be separate and after detection, it would invoke this flow. Anyway will try to introduce this `is_managed` flag with cluster_node_context object and use it.

Reply:
I still feel flow should be independent of detection code and as expected it should expect a nodes list for executing the expand of cluster. If that was the case we could just formed and saved jobs while detection itself, why have a separate flow defined for this?

Reply:
I am not talking about detecting the sds or making detected_cluster_id.  This flow will be called only after the new nodes have the same TendrlContext (i.e same integration_id).  So all you need to do is read list of nodes from here https://github.com/Tendrl/commons/blob/master/tendrl/commons/flows/import_cluster/__init__.py#L38

and then import those nodes for which ClusterNodeContext.is_managed is "no"